pytorch_lightning/metrics/classification/helpers.py

```diff
@@ -181,13 +181,19 @@ def _check_num_classes_ml(num_classes: int, is_multiclass: bool, implied_classes
 
 
 def _check_top_k(top_k: int, case: str, implied_classes: int, is_multiclass: Optional[bool], preds_float: bool):
-    if "multi-class" not in case or not preds_float:
-        raise ValueError(
-            "You have set `top_k` above 1, but your data is not (multi-dimensional) multi-class"
-            " with probability predictions."
-        )
+    if case == "binary":
+        raise ValueError("You can not use `top_k` parameter with binary data.")
+    if not isinstance(top_k, int) or top_k <= 0:
```

Comment:
just to be sure: this will be checked once for functional and twice for class metrics (because we also check on initialization)?

Reply:
Yes, true. Checking on initialization was suggested in the previous PR for accuracy, so I am using it here too